sdb/commands/stacks.py

```diff
@@ -216,86 +248,120 @@ def find_module_memory_segment(mod_name: str) -> Tuple[int, int]:
-        ) in task_states_lowercase and not args.tstate in state_shortcuts:
-            raise sdb.CommandError(
-                self.name,
-                "'{:s}' is not a valid task state (acceptable states: {:s})".
-                format(args.tstate, ", ".join(task_states)))
-
-        if args.module and Stacks.find_module_memory_segment(
-                args.module)[0] == -1:
-            raise sdb.CommandError(
-                self.name,
-                "module '{:s}' doesn't exist or isn't currently loaded".format(
-                    args.module))
-
-    def _call(self, objs: Iterable[drgn.Object]) -> Iterable[drgn.Object]:
-        # pylint: disable=too-many-locals
-        # pylint: disable=too-many-branches
-        # pylint: disable=too-many-statements
+    @staticmethod
+    def find_mod_offset(module: str) -> Tuple[int, int]:
+        mod_start, mod_size = Stacks.find_module_memory_segment(module)
+        assert mod_start != -1
+        mod_end = mod_start + mod_size
+        return mod_end, mod_start
 
+    def validate_context(self) -> None:
         #
-        # As the exception explains the code that follows this statement
-        # only works for linux kernel targets (crash dumps or live systems).
-        # When support for userland is added we can factor the kernel code
-        # that follows into its own function and switch to the correct
+        # This implementation only works for linux kernel targets
+        # (crash dumps or live systems). When support for userland is added we can
+        # refactor the kernel code into its own function and switch to the correct
         # codepath depending on the target.
         #
         if not sdb.get_target_flags() & drgn.ProgramFlags.IS_LINUX_KERNEL:
             raise sdb.CommandError(self.name,
                                    "userland targets are not supported yet")
-        self.validate_args(self.args)
+        self.validate_args()
+
+    def validate_args(self) -> None:
+        if self.args.function:
+            try:
+                func = sdb.get_object(self.args.function)
+            except KeyError:
+                raise sdb.CommandError(
+                    self.name,
+                    "symbol '{:s}' does not exist".format(self.args.function))
+            if func.type_.kind != drgn.TypeKind.FUNCTION:
+                raise sdb.CommandError(
+                    self.name,
+                    "'{:s}' is not a function".format(self.args.function))
 
-        #
-        # Resolve TSTATE shortcut and/or sanitize it to standard uppercase
-        # notation if it exists.
-        #
         if self.args.tstate:
-            if self.args.tstate in Stacks.TASK_STATE_SHORTCUTS:
-                self.args.tstate = Stacks.TASK_STATES[
-                    Stacks.TASK_STATE_SHORTCUTS[self.args.tstate]]
-            else:
-                self.args.tstate = self.args.tstate.upper()
+            self.args.tstate = Stacks.resolve_state(self.args.tstate)
```

Comment:
In general it is better to use local variables and not mutate `self.args` variables themselves as this can introduce unexpected behavior in error messages and other places where they are used. Can we use a local variable for tstate here? (the error output should still use `self.args.tstate`)

Reply:
So I can add a local variable, but the original implementation of stacks does mutate self.args.state. Are you suggesting that we re-work that entirely? 

I don't think it's inherently bad that the error message gives an upper case version of what was passed (all the options it suggestions for valid states are upper case).

Reply:
Discussed offline: I'm adding a new instance variable so we don't mutate what was passed in args.